metrics/web/docs/features/deep_links/01_deep_linking.md

```diff
@@ -142,3 +166,206 @@ history.replaceState(null, 'metrics', '$uri');
+    return MetricsRoutes.loading;
+  }
+
+  final routeName = pathSegments.first;
+
+  if (routeName == RouteName.login.value) {
+    return MetricsRoutes.login;
+  } else if (routeName == RouteName.dashboard.value) {
+    return MetricsRoutes.dashboard;
+  } else if (routeName == RouteName.projectGroups.value) {
+    return MetricsRoutes.projectGroups;
+  } else if (routeName == RouteName.debugMenu.value) {
+    return MetricsRoutes.debugMenu;
+  }
+
+  return MetricsRoutes.dashboard;
+}
+```
+
+To be able to create a specific route with the `query parameters` we should extract the route matching logic to a separate `_parseRoute` method. Then we can use the parsed `RouteConfiguration` and provide it with the `query parameters` using the `RouteConfiguration.copyWith` method.
+
+Consider the following code snippet that demonstrates this:
+```dart
+RouteConfiguration create(Uri uri) {
+  final pathSegments = uri?.pathSegments;
+
+  if (pathSegments == null || pathSegments.isEmpty) {
+    return MetricsRoutes.loading;
+  }
+
+  final routeName = pathSegments.first;
+  final route = _getRoute(routeName);
+
+  final queryParameters = uri.queryParameters;
+  
+  return route.copyWith(queryParameters: queryParameters);
+}
+
+RouteConfiguration _getRoute(String routeName) {
+  if (routeName == RouteName.login.value) {
+    return MetricsRoutes.login;
+  } else if (routeName == RouteName.dashboard.value) {
+    return MetricsRoutes.dashboard;
+  } else if (routeName == RouteName.projectGroups.value) {
+    return MetricsRoutes.projectGroups;
+  } else if (routeName == RouteName.debugMenu.value) {
+    return MetricsRoutes.debugMenu;
+  }
+
+  return MetricsRoutes.dashboard;
+}
+```
+
+#### RouteInformationParser
+The `Navigator 2.0` integration uses the `RouteConfiguration` to restore the URLs when the navigation state changes (e.g., when the user opens or closes a page). This restoration takes place in the `MetricsRouteInformationParser.restoreRouteInformation()` method. Consider this method's code:
+```dart
+@override
+RouteInformation restoreRouteInformation(RouteConfiguration configuration) {
+  if (configuration == null) return null;
+
+  return RouteInformation(location: configuration.path);
+}
+```
+
+As we can see, the current implementation does not restore the `query parameters` of a URL, but only the URL's `path`. 
+
+To restore the `RouteInformation` correctly, we should implement a `RouteConfiguration.toUrl()` method, that converts a `RouteConfiguration` instance to a URL location.
+```dart
+/// A class that represents the configuration of the route.
+class RouteConfiguration {
+  /// A path of this route that is used to create the application URL.
+  final String path;
+  
+  /// A query parameters of this route that is used to create the application URL.
+  final Map<String, dynamic> queryParameters;
+  
+  /// ... other fields declarations, constructors
+  
+  /// Converts this instance to a URL [String].
+  String toUrl() {
```

Comment:
Would it be better to name this method `getUrl` or `createUrl` since it creates a URL based on the `RouteConfiguration` instance, but not converts this instance into the URL? What do you think?